frontend/src/bundles/common/pages/budgets/budget-details/helpers/calculate-budget-details.helper.ts

```diff
@@ -0,0 +1,60 @@
+const calculateBudgetDetails = ({
+    amount,
+    spent,
+    startDate,
+    recurrence,
+}: {
+    amount: number;
+    spent: number;
+    startDate: string;
+    recurrence: string;
+}): { moneyLeft: number; canSpend: number; lastDate: string } => {
+    const startPeriod = new Date(startDate);
+    let endDate;
+    switch (recurrence) {
+        case 'daily': {
+            endDate = new Date(startPeriod.getTime() + 86_400_000);
+            break;
+        }
+        case 'weekly': {
+            endDate = new Date(startPeriod.getTime() + 604_800_000);
+            break;
+        }
+        case 'biweekly': {
+            endDate = new Date(startPeriod.getTime() + 1_209_600_000);
+            break;
+        }
+        case 'monthly': {
+            endDate = new Date(
+                startPeriod.getFullYear(),
+                startPeriod.getMonth() + 1,
+                startPeriod.getDate(),
+            );
+            break;
+        }
+        case 'yearly': {
+            endDate = new Date(
+                startPeriod.getFullYear() + 1,
+                startPeriod.getMonth(),
+                startPeriod.getDate(),
+            );
+            break;
+        }
+        default: {
+            endDate = startPeriod;
+            break;
+        }
+    }
+    const lastDate = new Date(endDate).toLocaleDateString('en-US', {
+        month: 'short',
+        day: '2-digit',
+        year: 'numeric',
+    });
+    const timeRemainingMs = endDate.getTime() - Date.now();
+    const daysRemaining = Math.ceil(timeRemainingMs / (1000 * 60 * 60 * 24));
```

Comment:
Magic numbers

Reply:
resolved in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/286/commits/b20760f84841bd84bb032c5ba118188e58a721e5](url)